File: go-pr/48/go/code-quality-naming/data_handler.go
package main

import "fmt"

type DataHandler struct{}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (dh *DataHandler) proc(d []int, x int) {
	for i := 0; i < len(d); i++ {
		d[i] = d[i] * x
	}
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (dh *DataHandler) MultiplyAllElements(numbers []int, multiplier int) {
	for index := 0; index < len(numbers); index++ {
		numbers[index] = numbers[index] * multiplier
	}
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (dh *DataHandler) chk(s string) bool {
	return len(s) > 0 && len(s) < 100
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (dh *DataHandler) IsValidStringLength(inputString string) bool {
	return len(inputString) > 0 && len(inputString) < 100
}
// {/fact}

func main() {
	handler := &DataHandler{}
	numbers := []int{1, 2, 3, 4, 5}
	handler.proc(numbers, 2)
	fmt.Println(numbers)
}

File: cpp-pr/46/cpp/code-quality-naming/data_handler.cpp
#include <iostream>
#include <vector>
#include <string>

class DataHandler {
public:
    // {fact rule=code-quality-naming@v1.0 defects=1}
    void proc(std::vector<int>& d, int x) {
        for (int i = 0; i < d.size(); i++) {
            d[i] = d[i] * x;
        }
    }
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=0}
    void multiplyAllElements(std::vector<int>& numbers, int multiplier) {
        for (size_t index = 0; index < numbers.size(); index++) {
            numbers[index] = numbers[index] * multiplier;
        }
    }
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=1}
    bool chk(const std::string& s) {
        return s.length() > 0 && s.length() < 100;
    }
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=0}
    bool isValidStringLength(const std::string& inputString) {
        return inputString.length() > 0 && inputString.length() < 100;
    }
    // {/fact}
};

int main() {
    DataHandler handler;
    std::vector<int> numbers = {1, 2, 3, 4, 5};
    handler.proc(numbers, 2);
    return 0;
}